packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -221,25 +222,39 @@ class Vif extends BaseComponent {
   resourceSets: subscribeResourceSets,
   user: subscribeCurrentUser,
 })
-@connectStore(() => ({
-  isAdmin: createSelector(
+@connectStore(() => {
+  const getIsAdmin = createSelector(
     getUser,
     user => user && user.permission === 'admin'
-  ),
-  isPoolAdmin: getIsPoolAdmin,
-  networks: createGetObjectsOfType('network').sort(),
-  pool: createGetObject((_, props) => props.location.query.pool),
-  pools: createGetObjectsOfType('pool'),
-  templates: createGetObjectsOfType('VM-template').sort(),
-  userSshKeys: createSelector(
+  )
+  const getNetworks = createGetObjectsOfType('network').sort()
+  const getPool = createGetObject((_, props) => props.location.query.pool)
+  const getPools = createGetObjectsOfType('pool')
+  const getSrs = createGetObjectsOfType('SR')
+  const getTemplates = createGetObjectsOfType('VM-template').sort()
+  const getUserSshKeys = createSelector(
     (_, props) => {
       const user = props.user
       return user && user.preferences && user.preferences.sshKeys
     },
     keys => keys
-  ),
-  srs: createGetObjectsOfType('SR'),
-}))
+  )
+  return (state, props) => ({
+    isAdmin: getIsAdmin(state, props),
+    isPoolAdmin: getIsPoolAdmin(state, props),
+    networks: getNetworks(state, props),
+    pool: getPool(state, props),
+    pools: getPools(state, props),
+    resolvedResourceSets: getResolvedResourceSets(
+      state,
+      props,
+      props.pool === undefined // to get objects as a self user
```

Comment:
Do we always want this for `getResolvedResourceSets`? If so, we may want to change `getResolvedResourceSets` itself. Maybe not in this PR, though.

Reply:
yes we need that to get the objects for self-user because `getResolvedResourceSets` get objects with `createGetObjectsOfType` and that's why we add `useResourceSet` in `_getPermissionsPredicate`

Reply:
Yes, but if we *always* need this with `getResolvedResourceSets`, we may want to implement the permission bypass directly in `getResolvedResourceSets`.